src/backend/distributed/worker/worker_drop_protocol.c

```diff
@@ -176,6 +169,27 @@ WorkerDropDistributedTable(Oid relationId)
 
 	/* delete the row from pg_dist_partition */
 	DeletePartitionRow(relationId);
+
+	if (!IsObjectAddressOwnedByExtension(&distributedTableObject, NULL))
+	{
+		/*
+		 * If the table is owned by an extension, we cannot drop it, nor should we
+		 * until the user runs DROP EXTENSION. Therefore, we skip dropping the
+		 * table.
+		 */
+		char *relName = get_rel_name(relationId);
+		Oid schemaId = get_rel_namespace(relationId);
+		char *schemaName = get_namespace_name(schemaId);
+
+		StringInfo dropCommand = makeStringInfo();
```

Comment:
maybe add a comment that we use ProcessUtility to make sure we also drop objects that depend on the table and call the event trigger which removes them from pg_dist_object.